stripe/http_client.py

```diff
@@ -134,6 +149,14 @@ def request_with_retries(self, method, url, headers, post_data=None):
                 time.sleep(sleep_time)
             else:
                 if response is not None:
+                    rheaders = response[2]
+                    if "Request-Id" in rheaders and stripe.enable_telemetry:
+                        request_id = rheaders["Request-Id"]
+                        request_duration_ms = _now_ms() - request_start
+                        self._set_last_request_metrics(
+                            RequestMetrics(request_id, request_duration_ms)
+                        )
```

Comment:
And maybe extract this one to something like `_store_telemetry(request_start, rheaders)`.

You may want to unwind `_last_request_metrics` and `_set_last_request_metrics` into the new helpers too.